zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -843,6 +865,11 @@ object Http {
 
   private case class Attempt[A](a: () => A) extends Http[Any, Nothing, Any, A]
 
+  private final case class Compose[R, E, EE, A, B, BB](
```

Comment:
```suggestion
  private final case class Combine[R, E, EE, A, B, BB](
```

Also this is faster than our current `++`.